cli/eva_sub_cli.py

```diff
@@ -1,20 +1,69 @@
 
+from cli import SUB_CLI_CONFIG_FILE, __version__
+from cli.docker_validator import DockerValidator, docker_path, container_image
 from cli.submit import StudySubmitter
 
+VALIDATION_OUTPUT_DIR = "validation_output"
+VALIDATE = 'validate'
+SUBMIT = 'submit'
+RESUME = 'resume'
+
+logging_config.add_stdout_handler()
+
+
+def get_docker_validator(vcf_files_mapping, output_dir, metadata_json, metadata_xlsx,
+                         arg_container, arg_docker, sub_config):
+    docker = arg_docker or docker_path
+    container = arg_container or container_image
+    validation_output_dir = os.path.join(output_dir, VALIDATION_OUTPUT_DIR)
+    return DockerValidator(vcf_files_mapping, validation_output_dir, metadata_json, metadata_xlsx,
+                           container, docker, sub_config)
+
+
 if __name__ == "__main__":
-    argparse = ArgumentParser(description='EVA Submission CLI')
-    argparse.add_argument('--submission-dir', required=True, type=str,
-                          help='Full path to the submission directory where all submission info is/will be stored')
-    argparse.add_argument('--resume', action='store_true', default=False, help='resume an existing submission')
-    args = argparse.parse_args()
-
-    logging_config.add_stdout_handler()
-
-    submitter = StudySubmitter()
-    if args.resume:
-        submitter.upload_submission(args.submission_dir)
-    else:
-        submitter.submit(args.submission_dir)
+    argparser = ArgumentParser(description='EVA SUB CLI - to validate and submit a submission')
+    argparser.add_argument('--task', required=True, choices=[VALIDATE, SUBMIT, RESUME],
+                           help='Select a task to perform')
+    argparser.add_argument('--submission_dir', required=True, type=str,
+                           help='Full path to the directory where all processing will be done and submission info is/will be stored')
+    argparser.add_argument("--docker_path", help="Full path to the docker installation, "
+                                                 "not required if docker is available on path", required=False)
+    argparser.add_argument("--container_name", help="Name of the docker container", required=False)
+    argparser.add_argument("--vcf_files_mapping", required=False,
+                           help="csv file with the mappings for vcf files, fasta and assembly report")
+    group = argparser.add_mutually_exclusive_group(required=False)
+    group.add_argument("--metadata_json",
+                       help="Json file that describe the project, analysis, samples and files")
+    group.add_argument("--metadata_xlsx",
+                       help="Excel spreadsheet  that describe the project, analysis, samples and files")
+
+    args = argparser.parse_args()
+
+    # load config
+    config_file_path = os.path.join(args.submission_dir, SUB_CLI_CONFIG_FILE)
+    sub_config = WritableConfig(config_file_path, version=__version__)
+
+    if args.task == RESUME:
```

Comment:
From a design standpoint, if validation fails for a submission, I feel that issuing "resume" at that point should resume the validation.

Reply:
I think I'd agree. We might want to discuss what is and what isn't resumable but if we have a generic resume option it should apply to all steps.

Reply:
This being said implementing a comprehensive resume strategy seems outside of the scope of this ticket.

Reply:
Tried to simplify resume option